### ISSUE TITLE
run.cmd - support explicit exec form

### ISSUE THREAD

**Issue by mgurov** [rhuss/docker-maven-plugin#253]
At the moment it is not possible to run a more complicated command on a container start, e.g. `docker run -it --rm busybox /bin/sh -c "while true; do echo Hello world; sleep 1; done"` as the cmd is split by spaces at  [ContainerCreateConfig.command](https://github.com/rhuss/docker-maven-plugin/blob/master/src/main/java/org/jolokia/docker/maven/access/ContainerCreateConfig.java#L35), and the following configuration: 

````xml
              <image>
                  <name>busybox</name>
                  <run>
                      <cmd>/bin/sh -c "while true; do echo Hello world; sleep 1; done"</cmd>
                      <wait>
                          <log>Hello world</log>
                      </wait>
                  </run>
              </image>
````
produces container startup error `true;: line 1: syntax error: unterminated quoted string`. 

It would be nice to have an ability to specify the exec form of the command explicitly, e.g.

````xml
              <image>
                  <name>busybox</name>
                  <run>
                      <cmd>
                          <params>
                              <param>/bin/sh</param>
                              <param>-c</param>
                              <param>while true; do echo Hello world; sleep 1; done</param>
                          </params>
                      </cmd>
                      <wait>
                          <log>Hello world</log>
                      </wait>
                  </run>
              </image>

````

in line with [recent changes to ENTRYPOINT/CMD](https://github.com/rhuss/docker-maven-plugin/commit/cc6f348324d958c2943cb25d91b2ab4cb8c4e1e )

**Comment by rhuss** [comment]
You are right, the command specified for `<run>` should support the same capabilities like in `<build>` i.e. specifying the shell and exec forms explicitely.

We will include this in one of the next section, if you like to speed this up you could help us with a PR: Starting point is the [RunConfiguration](https://github.com/rhuss/docker-maven-plugin/blob/integration/src/main/java/org/jolokia/docker/maven/config/RunImageConfiguration.java#L38) where the command should be switched over to [Arguments](https://github.com/rhuss/docker-maven-plugin/blob/integration/src/main/java/org/jolokia/docker/maven/config/Arguments.java), similar as in `BuildImageConfiguration` for `entryPoint` and `cmd`. It is used [here](https://github.com/rhuss/docker-maven-plugin/blob/integration/src/main/java/org/jolokia/docker/maven/service/RunService.java#L213).

**Comment by mgurov** [comment]
I have created a [branch with a draft implementation](https://github.com/rhuss/docker-maven-plugin/compare/master...mgurov:run-cmd-entrypoint-explicit-args). Specifying arguments explicitly works like a charm: 
````xml
                <env>
                  <BLAH>FOOE</BLAH>
                </env>
                <cmd>
                  <exec>
                    <arg>sh</arg>
                    <arg>-c</arg>
                    <arg>echo $BLAH</arg>
                  </exec>
                </cmd>
````
 but switching the `<shell>` from array to simple string in API json didn't work as I expected. I kept getting `System error: exec:...stat...no such file or directory`. Fell back on splitting by spaces. The following configuration will produce cmd json `["the", "shell", "way"]` 
````xml
            <image>
              <name>debian:jessie</name>
              <run>
                <entrypoint>
                  <shell>/bin/echo</shell>
                </entrypoint>
                <cmd><shell>the shell way</shell></cmd>
                <log>
                  <prefix>keep splitting by the spaces</prefix>
                </log>
              </run>
            </image>
````
I suggest decreasing the configuration clutter by allowing the "Inlined collection" notation for the explicit arguments: 

````xml
              <run>
                <entrypoint>
                  <arg>echo</arg>
                </entrypoint>
                <cmd>
                  <arg>inlined args</arg>
                  <arg>mean less clutter</arg>
                </cmd>
              </run>
````

and default property for the Arguments bean to be the shell form: 

````xml
              <run>
                <entrypoint>echo</entrypoint>
                <cmd>default string for simple commands</cmd>
              </run>

````

I plan to ponder bit more on why doesn't plain string cmd/entrypoint work against the docker API, will then polish the branch (validation/tests) and create a pull request in a week time at worst.

**Comment by mgurov** [comment]
Pull request sent: #269
